src/extralit_ocr/jobs.py

```diff
@@ -0,0 +1,193 @@
+except ImportError:
+    HAS_EXTRALIT_SERVER = False
+    _LOGGER.warning("Extralit server components not available, using default margins")
+
+
+def _get_document_margins(document_id: UUID) -> Optional[Tuple[int, int, int, int]]:
+    """
+    Fetch margins from document metadata in database.
+
+    Returns:
+        Tuple of (left, top, right, bottom) margins in PDF points, or None if not found
+    """
+    if not HAS_EXTRALIT_SERVER:
+        return None
+
+    try:
+        with SyncSessionLocal() as session:
+            document = session.query(Document).filter(Document.id == document_id).first()
+
+            if not document or not document.metadata_:
+                _LOGGER.info(f"No metadata found for document {document_id}")
+                return None
+
+            metadata = document.metadata_
+            if isinstance(metadata, dict):
+                analysis_metadata = metadata.get("analysis_metadata", {})
+            else:
+                analysis_metadata = getattr(metadata, "analysis_metadata", {}) or {}
+
+            if isinstance(analysis_metadata, dict):
+                layout_analysis = analysis_metadata.get("layout_analysis", {})
+            else:
+                layout_analysis = getattr(analysis_metadata, "layout_analysis", {}) or {}
+
+            # Check for estimated_margins directly in layout_analysis (correct structure)
+            if isinstance(layout_analysis, dict) and "estimated_margins" in layout_analysis:
+                estimated_margins = layout_analysis["estimated_margins"]
+
+                # Convert pixels to PDF points (multiply by 0.75)
+                if all(key in estimated_margins for key in ["left_px", "top_px", "right_px", "bottom_px"]):
+                    left = int(estimated_margins["left_px"] * 0.75)
+                    top = int(estimated_margins["top_px"] * 0.75)
+                    right = int(estimated_margins["right_px"] * 0.75)
+                    bottom = int(estimated_margins["bottom_px"] * 0.75)
+
+                    margins = (left, top, right, bottom)
+                    _LOGGER.info(f"Using document-specific margins for {document_id}: {margins}")
+                    return margins
+                else:
+                    _LOGGER.info(f"Incomplete margin data for document {document_id}")
+
+    except Exception as e:
+        _LOGGER.warning(f"Error retrieving margins for document {document_id}: {e}")
+
+    return None
+
+
+async def _download_file_from_s3(s3_url: str) -> bytes:
+    """Download file content from S3 using existing extralit-server patterns."""
+    if not HAS_EXTRALIT_SERVER:
+        raise RuntimeError("S3 operations require extralit_server components")
+
+    try:
+        s3_client = await files.get_s3_client()
+        return await files.download_file_content(s3_client, s3_url)
+    except Exception as e:
+        raise RuntimeError(f"Failed to download file from S3: {e}")
+
+
+def pymupdf_to_markdown_job(

```

Comment:
You also removed the `@jobs` decorator here. It's needed so the RQ worker in the `extralit-server` can call `extralit_ocr.jobs.pymupdf_to_markdown_job`